### PR TITLE
feature/add di

### PR DESCRIPTION
**Pull request by Gh0stF1ghter**

### CODE REVIEW

**Review by 5Ka-me — CHANGES_REQUESTED** [review]

File: API/Program.cs [hunk 1]
```diff
@@ -6,10 +6,12 @@
...
 
-builder.Services.AddControllers();
+services.AddControllers();
 
-builder.Services.AddEndpointsApiExplorer();
-builder.Services.AddSwaggerGen();
+services.AddEndpointsApiExplorer();
+services.AddSwaggerGen();
+
+services.AddReposDependencies();
```
Comment (5Ka-me) [inline API/Program.cs:14]: Move your custom DI methods to a separate file, like you did in DAL

**Review by 5Ka-me — CHANGES_REQUESTED** [review]

File: DAL/DI/ServicesConfiguration.cs [hunk 2]
```diff
@@ -9,4 +11,13 @@ public static class ServicesConfiguration
...
-}
+
+    public static void AddReposDependencies(this IServiceCollection services)
+    {
+        services.AddTransient<IClientRepository, ClientRepository>();
+        services.AddTransient<IManufacturerRepository, ManufacturerRepository>();
+        services.AddTransient<IVehicleClientHistoryRepository, VehicleClientHistoryRepository>();
+        services.AddTransient<IVehicleModelRepository, VehicleModelRepository>();
+        services.AddTransient<IVehicleRepository, VehicleRepository>();
+    }
```
Comment (5Ka-me) [inline DAL/DI/ServicesConfiguration.cs:22]: Make a single entry point to this class

File: BLL/DI/ServicesConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+using Microsoft.Extensions.DependencyInjection;
+
+namespace BLL.DI;
+public static class ServicesConfiguration
+{
+    public static void AddApplicationDependencies(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.ConfigureDbContext(configuration);
+
+        services.AddReposDependencies();
```
Comment (5Ka-me) [inline BLL/DI/ServicesConfiguration.cs:12]: Use single method in a DAL

**Review by 5Ka-me — CHANGES_REQUESTED** [review]

File: DAL/DI/ServicesConfiguration.cs [hunk 4]
```diff
@@ -7,6 +9,14 @@ namespace DAL.DI;
 
 public static class ServicesConfiguration
 {
-    public static void ConfigureDbContext(this IServiceCollection services, IConfiguration configuration) =>
+    public static void AddDalDependencies(this IServiceCollection services, IConfiguration configuration)
```
Comment (5Ka-me) [inline DAL/DI/ServicesConfiguration.cs:12]: Perhaps it would be better to name this like "AddDataAccessDependencies"

**Review by 5Ka-me — APPROVED** [review]

**Review by DarPers — APPROVED** [review]

File: BLL/DI/ServicesConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿using DAL.DI;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace BLL.DI;
```
Comment (Vlada25) [inline BLL/DI/ServicesConfiguration.cs:5]: Add empty line after `namespace BLL.DI;` (chtoto na dushnom &#128517;)

File: BLL/DI/ServicesConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+public static class ServicesConfiguration
+{
+    public static void AddApplicationDependencies(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddDataAccessDependencies(configuration);
```
Comment (Vlada25) [inline BLL/DI/ServicesConfiguration.cs:10]: I guess you can use arrow function here

**Review by MaliyOna — APPROVED** [review]

**Pull request merged**